pkg/nullio/nullio.go

```diff
@@ -1,16 +1,24 @@
 package nullio
 
 import (
+	"context"
 	"fmt"
 	"io"
 )
 
-// Reader is a struct that implements the io.Reader interface. Read does not return when called.
-type Reader struct{}
+// Reader is a struct that implements the io.Reader interface.
+// Read does not return when called until the context is done.
+type Reader struct {
+	Ctx context.Context
+}
 
 func (r Reader) Read(_ []byte) (int, error) {
-	// An empty select blocks forever.
-	select {}
+	if r.Ctx.Err() != nil {
+		return 0, io.EOF
+	}
+
+	<-r.Ctx.Done()
+	return 0, io.EOF
```

Comment:
I've read these lines (and also the previous implementation) multiple times and still don't get how reading actually works here. Maybe the intention of the whole Reader in the nullio package is not super clear to me.

Reply:
Ah, yes, that makes sense. Should we add your explanation to the code / some documentation? 🙏